services/121-service/src/registration/registrations.controller.ts

```diff
@@ -632,18 +698,25 @@ export class RegistrationsController {
   public async sendCustomTextMessage(
     @Body() body: SendCustomTextDto,
     @Paginate() query: PaginateQuery,
-    @Req() req,
+    @Req() req: ScopedUserRequest,
     @Param('programId', ParseIntPipe) programId: number,
-    @Query() queryParams, // Query decorator can be used in combi with Paginate decorator
+    @Query('dryRun') dryRun = 'false', // Query decorator can be used in combi with Paginate decorator
   ): Promise<BulkActionResultDto> {
-    const userId = req.user.id;
+    const userId = req.user?.id;
+
+    if (typeof userId === 'undefined') {
+      throw new HttpException(
+        'User is not authenticated',
+        HttpStatus.UNAUTHORIZED,
+      );
+    }
```

Comment:
Isn't this something that can be handled by a helper-method?
So that the code doesn't need to be copy-pasted in every method in every/this controller?

Or isn't it actually the responsibility/functionality of some Nest.js-middleware?
Or could it work similar to how the `ParsIntPipe` (see a few lines up) works? Like a `ValidUserPipe` or `AuthenticatedUserPipe`?

Reply:
Done